internal/cmd/run.go

```diff
@@ -47,17 +51,38 @@ func runCmd() *cobra.Command {
 					return err
 				}
 
-				for _, arg := range args {
-					block, err := lookupCodeBlock(blocks, arg)
-					if err != nil {
-						return err
+				if runAll {
+					for _, block := range blocks {
+						if !block.IsExcludedFromRunAll() {
+							if category != "" && category != block.Category() {
+								continue
+							}
+							runBlocks = append(runBlocks, block)
+						}
 					}
-
-					if err := replace(replaceScripts, block.Lines()); err != nil {
-						return err
+					if len(runBlocks) == 0 {
+						return errors.New("No tasks to execute with the category provided")
+					}
+				} else {
+					for _, arg := range args {
+						block, err := lookupCodeBlock(blocks, arg)
+						if err != nil {
+							return err
+						}
+
+						if err := replace(replaceScripts, block.Lines()); err != nil {
+							return err
+						}
+
+						runBlocks = append(runBlocks, block)
 					}
+				}
+			}
 
-					runBlocks = append(runBlocks, block)
+			if runAll {
+				err := confirmExecution(cmd, strconv.Itoa(len(runBlocks)), parallel)
```

Comment:
fyi its pretty uncommon to use `strconv.Itoa`, its much more common to use something like `fmt.Sprintf("%d")`

Generally `Sprint` is the "swiss army knife" of string formatting in golang

Reply:
Thank you I removed already